### PR TITLE
chore: funcs to create kustomize plugins

### PR DESCRIPTION
**Pull request by bartoszmajsak**
## Description

The actual use with resMap is inlined and moved to the caller.

This way we can construct plugins on demand and use them as building blocks instead.

## Merge criteria:

- [ ] Have a meaningful commit messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by ykaliuta** [comment]
I still would prefer to hide actual implementation of manifests manipulation under some more high level api where intentions are more clear (instead of hiding more configuration but exposing more implementation), but no problem to put my ACK if you want that way.

May be comments in the caller of what kind of plugins Create* are creating will be useful.

**Comment by bartoszmajsak** [comment]
> I still would prefer to hide actual implementation of manifests manipulation under some more high level api where intentions are more clear

I agree, we have WIP kustomize support for `Feature` stuff which is aiming to provide something between those lines. This is actually a small portion of this work I thought I would stack.

> May be comments in the caller of what kind of plugins Create* are creating will be useful.

I can do that, but need ☕ first :)

File: pkg/plugins/addLabelsplugin.go [hunk 1]
```diff
@@ -2,15 +2,21 @@ package plugins
...
 
-func ApplyAddLabelsPlugin(componentName string, resMap resmap.ResMap) error {
-	nsplug := builtins.LabelTransformerPlugin{
+// CreateAddLabelsPlugin creates a label transformer plugin that ensures resources
+// to which this plugin is applied will have the Open Data Hub common labels included.
+//
+// It has a following characteristics:
+//   - It adds labels to the "metadata/labels" path for all resource kinds.
+//   - It adds labels to the "spec/template/metadata/labels" and "spec/selector/matchLabels" paths
+//     for resources of kind "Deployment".
```
Comment (bartoszmajsak) [inline pkg/plugins/addLabelsplugin.go:17]: @ykaliuta is it what you were looking for?

**Review by ykaliuta — APPROVED** [review]

**Comment by ykaliuta** [comment]
@bartoszmajsak  probably, ossm-authz-mvp branch can be removed?

**Comment by bartoszmajsak** [comment]
> @bartoszmajsak  probably, ossm-authz-mvp branch can be removed? 

No, not at all. This is from where this small change came (was donated), but it has way more than this.

**Comment by ykaliuta** [comment]
> > @bartoszmajsak  probably, ossm-authz-mvp branch can be removed?
> 
> No, not at all. This is from where this small change came (was donated), but it has way more than this.

Shouldn't it be forked repository branch, not the common's one?

**Comment by bartoszmajsak** [comment]
> > > @bartoszmajsak  probably, ossm-authz-mvp branch can be removed?
> > 
> > 
> > No, not at all. This is from where this small change came (was donated), but it has way more than this.
> 
> Shouldn't it be forked repository branch, not the common's one?

It could, but it's more a process question than anything else. A while back we agreed on a feature branch for authorization capability work. If this is a hard rule for the team we can move it elsewhere. Having it in a forked repo adds one extra step of keeping `incubation` branch in sync, so I am not sure how much this is helping us be productive and focused on the right things.

**Review by zdtsw — COMMENTED** [review]
/lgtm

**Pull request merged**